From c1788a8b80fe272e09687a956621f9b3348c83a6 Mon Sep 17 00:00:00 2001
From: Gustaw Lewandowski <gustaw.lewandowski@intel.com>
Date: Mon, 15 Feb 2016 16:42:00 +0100
Subject: [PATCH 2/3] usb: gadget: rndis: rx_errors counter is incremented
 every time

rx_errors counter is incremented every time then rndis network
interface receives a packet, so number of rx_packets == rx_errors
because empty skb_buff is put into queue instead of being released

Change-Id: I2f11bd7cb6780530a78bfc67d48eef9f90c0e372
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-21873
Signed-off-by: Gustaw Lewandowski <gustaw.lewandowski@intel.com>
Reviewed-on: https://android.intel.com:443/470485
Reviewed-on: https://android.intel.com:443/475309
---
 drivers/usb/gadget/rndis.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/rndis.c b/drivers/usb/gadget/rndis.c
index 3a0d2f5..d215f01 100644
--- a/drivers/usb/gadget/rndis.c
+++ b/drivers/usb/gadget/rndis.c
@@ -1113,7 +1113,7 @@ int rndis_rm_hdr(struct gether *port,
 	if (num_pkts > rndis_ul_max_pkt_per_xfer_rcvd)
 		rndis_ul_max_pkt_per_xfer_rcvd = num_pkts;
 
-	skb_queue_tail(list, skb);
+	dev_kfree_skb_any(skb);
 	return 0;
 }
 EXPORT_SYMBOL(rndis_rm_hdr);
-- 
1.9.1

